test/asynchronous/test_encryption.py

```diff
@@ -2419,6 +2420,310 @@ async def test_05_roundtrip_encrypted_unindexed(self):
         self.assertEqual(decrypted, val)
 
 
+# https://github.com/mongodb/specifications/blob/527e22d5090ec48bf1e144c45fc831de0f1935f6/source/client-side-encryption/tests/README.md#25-test-lookup
+class TestLookupProse(AsyncEncryptionIntegrationTest):
+    @async_client_context.require_no_standalone
+    @async_client_context.require_version_min(7, 0, -1)
+    async def asyncSetUp(self):
+        await super().asyncSetUp()
+        encrypted_client = await self.async_rs_or_single_client(
+            auto_encryption_opts=AutoEncryptionOpts(
+                key_vault_namespace="db.keyvault",
+                kms_providers={"local": {"key": LOCAL_MASTER_KEY}},
+            )
+        )
+        await encrypted_client.drop_database("db")
+
+        key_doc = json_data("etc", "data", "lookup", "key-doc.json")
+        key_vault = await create_key_vault(encrypted_client.db.keyvault, key_doc)
+        self.addCleanup(key_vault.drop)
```

Comment:
addCleanup -> addAsyncCleanup

Also can we use the client_context client here to drop the entire database instead of just the keyvault? This will fix the `InvalidOperation: Cannot use AsyncMongoClient after close"` errors (which were introduced now that we close these clients).

Reply:
Alternatively we can just leave the data there and skip the cleanup altogether.

Reply:
Ah good catch, and yes! done!